### ISSUE TITLE
Discussion: Rebuilding packages when dependencies are updated.

### ISSUE THREAD

**Issue by cemkeylan** [kisslinux/kiss#126]
As you are aware, when shared libraries are updated, they change symbols and
names. These sometimes require some dependent packages to be rebuilt. This, right
now, can only be handled manually. Either the package maintainers update the release
number of the dependent package, or the user manually rebuilds the program after the
update given that he/she is aware that the shared libraries were a dependency for the program
they use, and the dependent package needs to be rebuilt.

Or it leads to breakage assuming the user wasn't aware that the library was a dependency
for the program they use.

I thought of adding a new dependency type (like make dependencies) to the package manager 
called `rebuild` dependencies which rebuilds the package when its `rebuild dependency` receives
an update. But I am not sure whether that's a good idea or not. Nonetheless, this could solve
the issue of manually tracking package rebuilds when shared libraries receive an update.

I would like to have your opinions about this issue.

**Comment by dylanaraps** [comment]
You'd just bump the second version field of anything needing a rebuild.

Example:

1. New Python version is pushed.
2. All second fields in dependent version files must be incremented.
3. 'kiss update' will build Python and then it's dependencies.

Using Python as an example again: 

If you have packaged something which needs a rebuild for Python, you 
should know to bump the second field of your package when Python is 
updated.

I don't think this is something we need to automate at all. 

As the maintainer of a package you should know its ins and outs. 
You should know the dependencies of the package and their relationship
with the package. You should know about its development and
upcoming changes (or breaks if relevant).

The same goes for changes like the upcoming GCC 10. All of your
packages should be tested and working with GCC 10 prior to its
release. GCC 10 has been sitting in testing for around a month now.

Basically, it's your responsibility to increment the second version field
when your package requires a rebuild against a new library version.
